sbi/inference/posteriors/direct_posterior.py

```diff
@@ -159,21 +171,27 @@ def log_prob(
             support of the prior, -∞ (corresponding to 0 probability) outside.
         """
         x = self._x_else_default_x(x)
+        condition_shape = self.posterior_estimator._condition_shape
+        try:
+            x = x.reshape(*condition_shape)
+        except RuntimeError as err:
+            raise ValueError(
+                f"Expected a single `x` which should broadcastable to shape \
+                  {condition_shape}, but got {x.shape}. For batched eval \
+                  see issue #990"
+            ) from err
 
         # TODO Train exited here, entered after sampling?
         self.posterior_estimator.eval()
 
         theta = ensure_theta_batched(torch.as_tensor(theta))
-        theta_repeated, x_repeated = match_theta_and_x_batch_shapes(theta, x)
```

Comment:
this is not needed anymore because it is handled by the `DensityEstimator`?

Reply:
Exactly.